winlogbeat/sys/event_test.go

```diff
@@ -168,21 +169,13 @@ func TestXML(t *testing.T) {
 	}
 }
 
+// Tests that control characters other than CR and LF are escaped
+// when the event is decoded.
 func TestInvalidXML(t *testing.T) {
-	eventXML := fmt.Sprintf(`
-<Event>
-  <UserData>
-    <Operation_ClientFailure xmlns='http://manifests.microsoft.com/win/2006/windows/WMI'>
-      <Id>{00000000-0000-0000-0000-000000000000}</Id>
-      <Message>じゃあ宇宙カウボーイ。。。%s</Message>
-    </Operation_ClientFailure>
-  </UserData>
-</Event>
-`, "\x1b")
-	_, err := UnmarshalEventXML([]byte(eventXML))
-	if !assert.NoError(t, err) {
-		assert.Equal(t, err.Error(), "XML syntax error on line 6: illegal character code U+001B")
-	}
+	evXML := strings.Replace(allXML, "%1", "&#xD;\n\x1b", -1)
```

Comment:
I think including other whitespace characters would be good too since `\t` is used a lot in event logs.

Reply:
What was the story with the `\r`? I see you had to use `&#xD;`. Is it because of https://www.w3.org/TR/REC-xml/#sec-line-ends?

Reply:
Yep, the XML parser was getting rid of it.